system/Config/Factories.php

```diff
@@ -31,15 +31,16 @@ class Factories
      * Store of component-specific options, usually
      * from CodeIgniter\Config\Factory.
      *
-     * @var array<string, array>
+     * @var array[]
+     * @phpstan-var array<string, array<string, string|bool|null>>
```

Comment:
Should these two be the same? I mean the vendor-specific annotation does not use its vendor-specific tags (like `class-string` or `list`) only the regular annotations so `@var` and `@phpstan-var` should have the same values.

Reply:
If all the major IDEs understand `array<string, array<string, string|bool|null>>`, 
`@phpstan-var` is not need.

Reply:
Be always assumed they do but honestly I never looked into it. I'm not even sure what you would call this notation, since it isn't officially PHPDoc: https://docs.phpdoc.org/3.0/guide/guides/types.html#Arrays

Reply:
`@var` only should be fine, the `@phpstan-var` only needed when it using very specific use case, eg: `non-empty-array` https://phpstan.org/r/554bcbfe-693f-47f0-8608-3cbacca36d85

Reply:
Is `@var array<string, array<string, string|bool|null>>` used with VS Code?

Reply:
Yes 

![Screen Shot 2022-10-27 at 19 34 35](https://user-images.githubusercontent.com/459648/198285573-f19cb92c-74f1-49e3-a912-73757b7ed768.png)

Reply:
Thank you!

Reply:
Removed `@phpstan-return`.